File: src/main/java/com/learnershi/rclasssocket/repository/PostItRepository.java
package com.learnershi.rclasssocket.repository;

import com.learnershi.rclasssocket.entity.PostIt;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Mono;

public interface PostItRepository extends ReactiveMongoRepository<PostIt, String> {
    Mono<PostIt> findByClassRoomIdAndTabIndexAndPageIndex(String classRoomId, Integer tabIndex, Integer pageIndex);
}

File: src/main/java/com/learnershi/rclasssocket/repository/PostItRepository.kt
package com.learnershi.rclasssocket.repository

import com.learnershi.rclasssocket.entity.PostIt
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Mono


interface PostItRepository : ReactiveMongoRepository<PostIt?, String?> {
    fun findByClassRoomIdAndTabIndexAndPageIndex(classRoomId: String?, tabIndex: Int?, pageIndex: Int?): Mono<PostIt?>
}